src/flakeModule.nix

```diff
@@ -4,178 +4,182 @@
   pick,
   winnow,
   harvest,
-}: {
-  inputs,
-  config,
-  options,
-  lib,
-  ...
 }: let
-  inherit
-    (builtins)
-    all
-    concatMap
-    head
-    isList
-    length
-    mapAttrs
-    tail
-    zipAttrsWith
-    ;
-  inherit
-    (lib)
-    getValues
-    mkOption
-    mkOptionType
-    showDefs
-    showOption
-    types
-    literalExpression
-    ;
-
-  # ensures we always end up with a list of lists
-  # if more than one definition
-  mergeHarvesterOption = loc: defs: let
-    list = getValues defs;
-  in
-    if length list == 1
-    then head list
-    else if all isList list
-    then
-      concatMap (l:
-        if isList l
-        then l
-        else [l])
-      list
-    else throw "Cannot merge definitions of `${showOption loc}', as at least one merge candidate is not a list of strings or a list of list of strings. Definition values:${showDefs defs}";
-
-  harvesterType = mkOptionType {
-    name = "harvester";
-    description = "harvest path(s) as a list of strings or a list of lists of strings";
-    descriptionClass = "noun";
-    merge = mergeHarvesterOption;
-    emptyValue = {value = [];};
+  blockTypes = types.BlockTypes "Block Types usable in the Standard flake-parts module";
```

Comment:
Also, open to any suggestions for more-appropriate/helpful messaging here.

Reply:
Upstream I went to great lenghts [here](https://github.com/paisano-nix/core/blob/f71a2db9414d66663c03a65ade97a9f353fb6d55/grow/default.nix#L70-L77) and [here](https://github.com/divnix/std/blob/112cb5d025e344f58e3a4446c4bf034e318ea1bb/flake.nix#L64-L71) (taking great care to not sabotage `unsaceGetAttrsPos`), in order to produce a good `origin` value in `(BlockTypes "${origin "cellBlocks"} - grow[On]:cellBlocks" cellBlocks))`